plansys2_executor/src/plansys2_executor/BTBuilder.cpp

```diff
@@ -709,31 +731,46 @@ BTBuilder::t(int level)
   return ret;
 }
 
+void replace(std::string & str, const std::string & from, const std::string & to)
+{
+  size_t start_pos = std::string::npos;
+  while ((start_pos = str.find(from)) != std::string::npos) {
+    str.replace(start_pos, from.length(), to);
+  }
+}
+
 std::string
 BTBuilder::execution_block(const GraphNode::Ptr & node, int l)
 {
   const auto & action = node->action;
   std::string ret;
+  std::string ret_aux = bt_action_;
   const std::string action_id = "(" + action.action->name_actions_to_string() + "):" +
     std::to_string(static_cast<int>(action.time * 1000));
 
-  ret = ret + t(l) + "<Sequence name=\"" + action_id + "\">\n";
 
+  std::string wait_actions;
   for (const auto & previous_node : node->in_arcs) {
     const std::string parent_action_id = "(" +
       previous_node->action.action->name_actions_to_string() + "):" +
       std::to_string(static_cast<int>( previous_node->action.time * 1000));
-    ret = ret + t(l + 1) + "<WaitAtStartReq action=\"" + parent_action_id + "\"/>\n";
+    wait_actions = wait_actions + t(1) + "<WaitAtStartReq action=\"" + parent_action_id + "\"/>";
+
+    if (previous_node != *node->in_arcs.rbegin()) {
+      wait_actions = wait_actions + "\n";
+    }
   }
 
-  ret = ret + t(l + 1) + "<ApplyAtStartEffect action=\"" + action_id + "\"/>\n";
-  ret = ret + t(l + 1) + "<ReactiveSequence name=\"" + action_id + "\">\n";
-  ret = ret + t(l + 2) + "<CheckOverAllReq action=\"" + action_id + "\"/>\n";
-  ret = ret + t(l + 2) + "<ExecuteAction action=\"" + action_id + "\"/>\n";
-  ret = ret + t(l + 1) + "</ReactiveSequence>\n";
-  ret = ret + t(l + 1) + "<CheckAtEndReq action=\"" + action_id + "\"/>\n";
-  ret = ret + t(l + 1) + "<ApplyAtEndEffect action=\"" + action_id + "\"/>\n";
-  ret = ret + t(l) + "</Sequence>\n";
+  replace(ret_aux, "ACTION_ID", action_id);
```

Comment:
couldn't you just use `std::replace` from `<algorithm>` here instead of writing your own function?
https://stackoverflow.com/questions/2896600/how-to-replace-all-occurrences-of-a-character-in-string
https://www.cplusplus.com/reference/algorithm/replace/

Reply:
Absolutely!!

Reply:
Hi @xydesa 

I have just checked that this function only replaces one char :(